src/content/contribute/writing-a-scaffold.md

```diff
@@ -0,0 +1,387 @@
+---
+title: Writing a Scaffold
+sort: 3
+contributors:
+  - evenstensberg
+  - pranshuchittora
+  - EugeneHlushko
+  - misterdev
+  - rishabh3112
+---
+
+Welcome to the demonstration of the new `webpack init` command! To view what we are building today, run `webpack init webpack-scaffold-demo`. This demo will show you how to build your own webpack scaffold. Let's start by creating a file named `generator.js`.
+
+## Create a basic structure
+
+Before writing a `webpack-cli` scaffold, think about what you're trying to achieve. Do you want a "general" scaffold that could be used by any project or type of app? Do you want something focused, like a scaffold that writes both your `webpack.config.js` and your framework code? It's also useful to think about the user experience for your scaffold.
+
+`webpack-cli` offers interactive experience to customize the output accordingly. For example asking questions like: "What is your entry point?".
+
+Let's create our skeleton. In order for the webpack CLI to detect our options, we have to define some properties in the constructor.
+
+__generator.js__
+
+```js
+const Generator = require('yeoman-generator');
+
+module.exports = class WebpackGenerator extends Generator {
+  constructor(args, opts) {
+    super(args, opts);
+    opts.env.configuration = {
+      dev: {}
+    };
+  }
+};
+```
+
+`configuration` object has to have one property you name (we named it `dev` in the snippet above). A good practice is to name the underlying property with the name you want to give to your `webpack.config.js` file for a better indication of what configuration each file has.
+
+## Make it interactive
+
+In order for us to interact with the users, we make use of the [`prompting`](http://yeoman.io/authoring/user-interactions.html) method yeoman has. In this method, we can get various answers from the user, like asking for entry points or plugins. You can either manually create each object representing a question or you can make good use of our utilities from [`webpack-scaffold`](https://github.com/webpack/webpack-cli/tree/master/packages/webpack-scaffold). We are in a good mood today, so let's build a configuration only if the user chooses `Pengwings`.
```

Comment:
FYI there's a joke behind naming this pengwings: 

https://www.youtube.com/watch?v=9GHPNKUMf70

Reply:
Yeah i know that one :D

Reply:
:rofl: